CONTRIBUTING.md

```diff
@@ -0,0 +1,77 @@
+
+### Committing
+
+* Rebase your commits generously. Avoid having to open pull requests with a massive amount of commits, which all essentially work towards the same issue.
+* If possible, try to include all of the code, tests and documentation for one issue, into one commit.
+* The commit title should be in imperative form. Examples:
+  * Implement ...
+  * Add ...
+  * Change ...
+* The commit title should only state what you did. This helps keep commit titles short and easy to find.
+* The commit description should explain why you did it. Most times, referring to the issue itself is enough. Please do so with keywords (`fixes`, `resolves` etc.) in order to automatically link issues to commits. This helps to retrace commits and it simplifies opening pull requests.
+* Do not elaborate on how you did what you did - the code should be self-explanatory in this case. If the code is not self-explanatory enough (due to performance shortcuts or the sort), please explain this in comments, rather than in the commit.
+
+### Testing
+
+* One test should test one thing. Test code is still code, and so we should try to separate concerns and make it as readable as possible. This rule may be broken in cases where it does not make sense to separate concerns.
+* If there are internal and external tests related to the package, then mark the files with name\_in\_ test.go and name\_ex\_test.go
+* Make use of the `testify` package where possible. This clearly separates test code from production code, since the error handling is noticably different. It also provides useful functionality for checking equality, making sure certain fields are empty/not empty etc.
+
+### Logging
+
+* Use `github.com/sirupsen/logrus` for logging rather than `fmt`
+* As a general rule use:
+  * `log.Traceln` \(lowest level debug info\) mostly without dynamic parameters to debug flow \(use rarely and only when testing\)
+  * `log.Debugln` \(lower level debug info\) mostly with dynamic parameters to debug process a process \(use sparingly\)
+  * `log.Infoln` \(useful info for integration\) mostly with dynamic parameters to for general logging \(use sparingly\)
+  * `log.Warnln` \(situations potentially leading to errors or lower priority error detection\) always with dynamic parameters \(use at will\)
+  * `log.Errorln` \(recoverable errors\) always with dynamic parameters \(almost never used\)
+  * `log.Fatalln` \(always before panic\) always with dynamic parameters \(almost never used\)
+* Set dynamic parameters in the fields, not in the message. For instance set parameters in the following way:
+
+```text
+log.WithFields(log.Fields{ "param1" : param1, "param2": param2 }).Infoln("Some message")
+```
+
+Do not do the following:
+
+```text
+log.Infoln("Some message with params1 %v and param2 %v", param1, param2)
+```
+
```

Comment:
Some recommendation should be added about a logging template according to what the monitoring expects. I remember to have seen some log lines pushed to our infrastructure and laid out in a way that were easily searcheable by Logstash